x-pack/plugins/secops/public/components/range_date_picker/quick_select_popover/timer.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiButton,
+  EuiFieldNumber,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiSelect,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import React from 'react';
+import { pure } from 'recompose';
+
+interface Options {
+  value: string;
+  text: string;
+}
+
+export const singleLastOptions: Options[] = [
+  {
+    value: 'minutes',
+    text: 'minute',
+  },
+  {
+    value: 'hours',
+    text: 'hour',
+  },
+];
+
+export const pluralLastOptions: Options[] = [
+  {
+    value: 'minutes',
+    text: 'minutes',
+  },
+  {
+    value: 'hours',
+    text: 'hours',
+  },
+];
+
+interface Props {
+  timerIsOn: boolean;
+  duration: number;
+  durationKind: string;
+  onChange: (
+    stateType: string,
+    args: React.FormEvent<HTMLInputElement> | React.ChangeEvent<HTMLSelectElement>
+  ) => void;
+  toggleTimer: (timerIsOn: boolean) => void;
+}
+
+export const Timer = pure<Props>(({ onChange, timerIsOn, duration, durationKind, toggleTimer }) => (
+  <>
+    <EuiTitle size="xxxs">
+      <span>Refresh every</span>
+    </EuiTitle>
+    <EuiSpacer size="s" />
+    <EuiFlexGroup gutterSize="s" alignItems="center" responsive={false}>
+      <EuiFlexItem>
+        <EuiFormRow>
+          <EuiFieldNumber
+            aria-label="Count of"
+            value={duration}
+            step={0}
+            onChange={arg => {
+              onChange('duration', arg);
+            }}
+          />
+        </EuiFormRow>
+      </EuiFlexItem>
+      <EuiFlexItem>
+        <EuiFormRow>
+          <EuiSelect
+            aria-label="Count Type"
+            value={durationKind}
+            options={duration === 1 ? singleLastOptions : pluralLastOptions}
+            onChange={arg => {
+              onChange('durationKind', arg);
+            }}
+          />
+        </EuiFormRow>
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <EuiFormRow>
+          <EuiButton
+            iconType={timerIsOn ? 'stop' : 'play'}
+            onClick={toggleTimer.bind(null, !timerIsOn)}
```

Comment:
I would rather use an anonymous function over bind to pass arguments personally as it's clearer.